### PR TITLE
TEST/JENKINS: Unite devel and release tests, to reduce compilation time

### PR DESCRIPTION
**Pull request by yosefe**
## Why
Optimize testing time. Needed mostly for embedded systems such as BlueField.

### CODE REVIEW

File: contrib/test_jenkins.sh [hunk 1]
```diff
@@ -114,6 +114,26 @@ make_clean() {
...
 
+#
+# Configure and build
+#   $1 - mode (devel|release)
+#
+build() {
+	mode=$1
+	shift
+
+	config_args="--prefix=$ucx_inst --without-java"
```
Comment (yosefe) [inline contrib/test_jenkins.sh:125]: java is tested separately; no need to waste time on maven packages fetch

File: contrib/test_jenkins.sh [hunk 2]
```diff
@@ -169,14 +190,7 @@ unload_cuda_env() {
...
 	set +x
-	task=$1
-	ntasks=$2
-	tasks_per_worker=$(( (ntasks + nworkers - 1) / nworkers ))
-	my_tasks_begin=$((tasks_per_worker * worker))
-	my_tasks_end=$((my_tasks_begin + tasks_per_worker))
-
-	# set return value to 0 (success) iff ($my_tasks_begin <= $task < $my_tasks_end)
-	[ $task -ge $my_tasks_begin ] && [ $task -lt $my_tasks_end ]
+	[[ $((task % nworkers)) -eq ${worker} ]]
```
Comment (yosefe) [inline contrib/test_jenkins.sh:193]: for more balanced tasks distribution

File: contrib/test_jenkins.sh [hunk 3]
```diff
@@ -225,6 +239,7 @@ get_my_tasks() {
 get_ib_devices() {
 	state=$1
 	device_list=$(ibv_devinfo -l | tail -n +2)
+	set +x
```
Comment (yosefe) [inline contrib/test_jenkins.sh:242]: avoid excess output

File: contrib/test_jenkins.sh [hunk 4]
```diff
@@ -347,21 +363,6 @@ prepare() {
 	cd build-test
 }
 
-check_make_distcheck() {
```
Comment (yosefe) [inline contrib/test_jenkins.sh:350]: was unused

**Comment by yosefe** [comment]
@hoopoepg @brminich can you pls take a look?

**Review by hoopoepg — APPROVED** [review]

**Review by brminich — APPROVED** [review]

**Pull request merged**